src/Recurrence.ts

```diff
@@ -212,4 +207,149 @@ export class Recurrence {
 
         return this.toText() === other.toText(); // this also checks baseOnToday
     }
+
+    /**
+     * nextAfter returns the next occurrence's date after `after`, based on the given rrule.
+     *
+     * The common case is that `rrule.after` calculates the next date and it
+     * can be used as is.
+     *
+     * In the special cases of monthly and yearly recurrences, there exists an
+     * edge case where an occurrence after the given number of months or years
+     * is not possible. For example: A task is due on 2022-01-31 and has a
+     * recurrence of `every month`. When marking the task as done, the next
+     * occurrence will happen on 2022-03-31. The reason being that February
+     * does not have 31 days, yet RRule sets `bymonthday` to `31` for lack of
+     * having a better alternative.
+     *
+     * In order to fix this, `after` will move into the past day by day. Each
+     * day, the next occurrence is checked to be after the given number of
+     * months or years. By moving `after` into the past day by day, it will
+     * eventually calculate the next occurrence based on `2022-01-28`, ending up
+     * in February as the user would expect.
+     */
+    private nextAfter(after: Moment, rrule: RRule): Date {
+        // We need to remove the timezone, as rrule does not regard timezones and always
+        // calculates in UTC.
+        // The timezone is added again before returning the next date.
+        after.utc(true);
+        let next = window.moment(rrule.after(after.toDate()));
+
+        // If this is a monthly recurrence, treat it special.
+        const monthMatch = this.toText().match(/every( \d+)? month(s)?(.*)?/);
+        if (monthMatch !== null) {
+            // ... unless the rule fixes the date, such as 'every month on the 31st'
+            if (!this.toText().includes(' on ')) {
```

Comment:
Instead of invoking `toText` again, you could use the eixisting `monthMatch` to find out if the part after `months` includes ` on `. Not a big deal, though.

Reply:
Thank you.

Reply:
I tried doing that, but because there is not always a part of months, it got a bit fiddly.

So instead I extracted a variable for the first `this.toText()` and re-used it: 24862ba5186ee299e56367a374e2d3e886e516ea